Riot/Modules/Authentication/Views/TermsView.swift

```diff
@@ -0,0 +1,208 @@
+        super.init(frame: frame)
+        loadNibContent()
+        commonInit()
+    }
+
+    private func commonInit() {
+
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.separatorStyle = .none
+        tableView.register(TableViewCellWithCheckBoxAndLabel.nib(), forCellReuseIdentifier: TableViewCellWithCheckBoxAndLabel.defaultReuseIdentifier())
+
+        acceptButton.layer.cornerRadius = 5
+        acceptButton.clipsToBounds = true
+        acceptButton.setTitle(NSLocalizedString("accept", tableName: "Vector", comment: ""), for: .normal)
+        acceptButton.setTitle(NSLocalizedString("accept", tableName: "Vector", comment: ""), for: .highlighted)
+
+        customizeViewRendering()
+    }
+
+    func customizeViewRendering() {
+        acceptButton.backgroundColor = kRiotColorGreen
+    }
+
+
+    // MARK: - Public
+
+    /// Display a list of policies the end user must accept.
+    ///
+    /// - Parameters:
+    ///   - terms: Terms data sent by the homeserver.
+    ///   - onAccepted: block called when the user has accepted all of them.
+    @objc func displayTerms(terms: MXLoginTerms, onAccepted: @escaping () -> Void) {
+
+        acceptedCallback = onAccepted
+
+        let lang: String? = Bundle.mxk_language()
+
+        policies = terms.policiesData(forLanguage: lang, defaultLanguage: "en")
+        acceptedPolicies.removeAll()
+
+        reload()
+    }
+
+
+    // MARK: - Private
+
+    private func reload() {
+        tableView.reloadData()
+
+        // Enable the button only if the user has accepted all policies
+        acceptButton.isEnabled = (policies.count == acceptedPolicies.count)
+        acceptButton.alpha = acceptButton.isEnabled ? 1 : 0.5
+    }
+
+    @IBAction func didAcceptButtonTapped(_ sender: Any) {
+        if policies.count == acceptedPolicies.count {
+            acceptedCallback?()
+        }
+    }
+
+
+    // MARK: - UITableViewDataSource
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return policies.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+
+        let cell = tableView.dequeueReusableCell(withIdentifier: TableViewCellWithCheckBoxAndLabel.defaultReuseIdentifier(), for: indexPath) as! TableViewCellWithCheckBoxAndLabel
+
+        let policy = policies[indexPath.row]
+        let accepted = acceptedPolicies .contains(indexPath.row)
+
+        cell.label.text = policy.name
+        cell.isEnabled = accepted
+        cell.accessoryType = .disclosureIndicator
+
+
+        let gesture: UITapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(didTapCheckbox))
+        gesture.numberOfTapsRequired = 1
+        gesture.numberOfTouchesRequired = 1
+
+        cell.checkBox.tag = indexPath.row
+        cell.checkBox?.isUserInteractionEnabled = true
+        cell.checkBox?.addGestureRecognizer(gesture)
```

Comment:
Check if a UITapGestureRecognizer is already present to avoid adding several gesture recognizers